apimonitor/views.py

```diff
@@ -119,6 +122,71 @@ def get_queryset(self):
         queryset = APIMonitor.objects.filter(user=self.request.user)
         return queryset
     
+    def retrieve_with_param(last_chosen_period_in_hours, n_bar, increment_in_minutes, monitor, success_rate, response_time):
```

Comment:
One code smell on this function. Bisa coba direname "last_chosen_period_in_hours" to "self" or add the missing "self" parameter?